### PR TITLE
Feature update: UseWindowSize hook.

### PR DESCRIPTION
**Pull request by francoisdillinger**
# Pull Request Description

**Description**
A hook that provides the current window dimensions.

**Acceptance Criteria**
- [x] **Dimensions:** Returns the current width and height of the window.
- [x] **Cleanup:** Removes resize event listeners when the component is unmounted.

**Details:**
- The `useWindowSize` hook returns the current window width and height.
- The values update dynamically when resizing the window.
- The resize event listener is properly cleaned up when the component is unmounted.
- Updated the `index.ts` file to export the `useWindowSize` hook.
- Updated the `docs/README` with a description of the new hook.

**Demo:**

Here is a gif demonstrating the hook in action:

useWindowSize Hook Demo

### Code Changes

**Hook Implementation:**
```javascript
import { useState, useEffect } from 'react';

export function useWindowSize() {
  const [windowSize, setWindowSize] = useState({
    width: window.innerWidth,
    height: window.innerHeight,
  });

  useEffect(() => {
    const handleResize = () => {
      setWindowSize({
        width: window.innerWidth,
        height: window.innerHeight,
      });
    };

    window.addEventListener('resize', handleResize);

    return () => window.removeEventListener('resize', handleResize);
  }, []);

  return windowSize;
}
```

**Index File Update:**
```typescript
export { useWindowSize } from './hooks/useWindowSize';
// ...other exports
```

**Documentation Update:**
Added a description of the `useWindowSize` hook to the `docs/README` file.
```
- [`useWindowSize()`](https://reacthaiku.dev/docs/hooks/useWindowSize) - Use hook to get width and height of window. Values update upon window resizing.
```

I wasn't sure if contributors were supposed to edit the `/docs` folder and implement the code for the demos or not so I decided to err on the side of caution and leave that alone. If there is anything else I need to do before this contribution is accepted, please let me know. THanks.

### CODE REVIEW

File: lib/hooks/useWindowSize.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from "react";
+
+export const useWindowSize = () => {
```
Comment (DavidHDev) [inline lib/hooks/useWindowSize.ts:3]: You don't seem to be leveraging Typescript in this implementation, what I would do is add a type and use it throughout the file:

```typescript
// Define the type for window size
type WindowSize = {
    width: number;
    height: number;
};

export const useWindowSize = (): WindowSize => {
    // Use type annotations for state
    const [windowSize, setWindowSize] = useState<WindowSize>({
        width: window.innerWidth, 
        height: window.innerHeight
    });

// rest of the implementation
```

File: lib/hooks/useWindowSize.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    const [windowSize, setWindowSize] = useState({
+        width: window.innerWidth, 
+        height: window.innerHeight
+    });
+
+    useEffect(() =>{
+        const handleResize = () => {
```
Comment (DavidHDev) [inline lib/hooks/useWindowSize.ts:10]: Nitpicky, but let's keep all the line indentation at `2 spaces`. It's like that throughout all the files

**Comment by DavidHDev** [comment]
PS: if you want to, you are free to update the `/docs` as well, just start the project locally and make sure the new features are documented in a similar way to the existing ones 🙏

File: lib/hooks/useWindowSize.ts (hunk 1 above)
Comment (francoisdillinger) [reply lib/hooks/useWindowSize.ts:8]: Annotations have been added.

File: lib/hooks/useWindowSize.ts (hunk 2 above)
Comment (francoisdillinger) [reply lib/hooks/useWindowSize.ts]: Fixed spacing as well.

File: lib/hooks/useWindowSize.ts (hunk 1 above)
Comment (francoisdillinger) [reply lib/hooks/useWindowSize.ts]: Return type also added.

**Review by DavidHDev — APPROVED** [review]

**Pull request merged**